transport/src/main/java/io/netty/channel/AdaptiveRecvByteBufAllocator.java

```diff
@@ -84,14 +81,14 @@ private static int getSizeTableIndex(final int size) {
         }
     }
 
-    private static final class HandleImpl implements Handle {
+    private final class HandleImpl extends MaxMessageHandle {
         private final int minIndex;
         private final int maxIndex;
         private int index;
         private int nextReceiveBufferSize;
         private boolean decreaseNow;
 
-        HandleImpl(int minIndex, int maxIndex, int initial) {
+        public HandleImpl(int minIndex, int maxIndex, int initial) {
```

Comment:
The class is `private` however there is no reason why the visibility of this method should be restricted.